drivers/usb/device/usb_dc_stm32.c

```diff
@@ -846,12 +846,9 @@ int usb_dc_ep_mps(const u8_t ep)
 
 int usb_dc_detach(void)
 {
-	// stop usb dev
 	HAL_PCD_Stop(&usb_dc_stm32_state.pcd);
```

Comment:
We don't need do this as `HAL_PCD_DeInit()` calls `HAL_PDC_Stop()`

Reply:
so let's break it down to only the `HAL_PCD_DeInit` call?
What is with the endpoints?

Reply:
> What is with the endpoints?

Probably application needs to close all the endpoints that it opened. As far as I understand, it depends when `usb_disable()` is called. 

I am searching out STM32 documentation for the correct sequence to disable USB device. So far no luck. @erwango Could you help on this?

Reply:
> @StefJar How did you test the change? Which sample?
> In PR #16026, when HAL_PCD_DeInit() was called, the application got stuck. Did you face a similar problem?
> @jli157 FYI.

nope
it worked for me fine(on a stm32f412)

Reply:
> it worked for me fine(on a stm32f412)

How did you test it? Which sample or test case?

Reply:
I have a sequence to put the device into sleep mode/reset. Inside that sequence I am deinit the usb. This works like follows:
- call `usb_disable`
- unint pins via `stm32_setup_pins`
- unclock usb via `LL_APB2_GRP1_DisableClock(LL_AHB2_GRP1_PERIPH_OTGFS)`

as you can see it is more less handcrafted. The test was run on 5 units (stm32f412) and went well